custom-vu/src/main/kotlin/jces1209/vu/page/PageLoader.kt

```diff
@@ -0,0 +1,81 @@
+package jces1209.vu.page
+
+import org.openqa.selenium.By
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions.or
+import org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated
+import com.atlassian.performance.tools.jiraactions.api.page.wait
+import org.openqa.selenium.support.ui.ExpectedCondition
+import java.time.Duration
+
+class PageLoader private constructor(
```

Comment:
I usually stray away from abstract names. Especially in page objects. If I cannot point to a screen and show it, then it's too abstract. When does a user see a page loader?
Perhaps, this represents a generic `class Page`, which has `expectedContent: List<By>` and `potentialErrors: List<By>`.

Reply:
I like that naming scheme!

Reply:
Over the weekend I had a thought: overly generic names are also bad, because they attract random, unrelated behavior. `Page` is way too open, sounds like a god class. It's current responsibility is to encapsulate the pattern of unhappy paths on a page. Following [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle), one responsibility is the max, so let's limit the concept via naming, e.g. `ErroringPage`, `FailingPage`, `UnhappyPage`, `BrittlePage` or something else conveying the meaning of "a page that can emit errors".